modules/apps/frontend-token/frontend-token-definition-api/src/main/java/com/liferay/frontend/token/definition/FrontendTokenDefinition.java

```diff
@@ -23,4 +25,7 @@
 
 	public String getJSON(Locale locale);
 
+	public ParsedFrontendTokenDefinition getParsedFrontendTokenDefinition(
```

Comment:
Couldn't we just expose the `getCategories`, `getSets`,  and `getTokens` collection methods and avoid exposing this weird `ParsedTokenDefinition` class?

I would go as far as to not create as an API unless we're planning to mutate the definition in java. I guess it could be useful at that point, but it feels weird for now and we should consider other things if we want to go there.

Reply:
you mean limiting the API and not exposing the whole frontend-token-definition schema?

Reply:
or simply removing `ParsedFrontendTokenDefinition` and putting its methods in `FrontendToken` directly?

I created a `ParsedFrontendTokenDefinition` because it implies a parsing operation which is more costly than accessing the JSON directly, but thinking of it again, it is maybe better to remove it...

Reply:
I didn't check the implementation, so I guess you're only parsing the first time it is requested and then caching it?

I don't know how expensive the parsing will be... I was thinking that if we payed the cost upfront, when creating the `FrontenTokenDefinition`, then we wouldn't need to have the `ParsedFrontendTokenDefinition` class and simply expose the fields and accessors we think are valuable from the schema.

Reply:
> simply expose the fields and accessors we think are valuable from the schema.

That is my doubt... 

I have a commit (locally) to expose the full schema to avoid future small PRs to add one tiny thing that will increase the API version and cause overhead. However, it only makes sense if the JSON schema (as it is now) is somewhat final, which I think it is, as this has been released and we have already exposed a public method to retrieve it.

So, is that the case? Can we consider the JSON schema stable enough to release the API now?

Reply:
> Can we consider the JSON schema stable enough to release the API now?

I don't think so.

Additions, though would only cause minor increments, so the overhead should be relatively low. Thus, I'd rather add the minimal API surface possible now (categories, sets, tokens, mappings) and then add more as needed.

One option would be for each entity to also have a `.toJSON()` so even if we miss something, it could be parsed by client code until a proper API is added.

Reply:
:thinking: We already give the full JSON, but it's true that a toJSON() method would be handy (at least you would have the subpart of JSON parsed for you). Let's go with that for now.

Regarding the minimal surface possible, this PR implements that, so I'll push tests for it and stop modeling the rest of the schema.

@victorg1991 do you foresee any other method you will need for sure in the future? (you can see what we have now in [this commit](https://github.com/liferay-frontend/liferay-portal/pull/412/commits/b8bc9a0b42120def2cec7ac0d3cff854310c1223))

Reply:
One more thing, now that I'm modifying the code... We cannot get rid of `ParsedFrontendTokenDefinition` because when parsing we need to define the `Locale` but the raw `FrontendTokenDefinition` is not assigned to any :-(.

However, I propose calling it `LocalizedFrontendTokenDefinition` instead. What do you think @jbalsas ?

Reply:
Another option would be propagating the `Locale` parameter to any method that needs it. However that makes the API and the implementation more complex... :thinking:

Reply:
I think it could be useful to expose name and label also for the frontend tokens at least, any reason to not do so? Also I think exposing something like getCssVariableName can also be very convenient, in order to avoid looping in the mappings array, what do you think?

Reply:
Hey @victorg1991, could you provide a quick recollection of links to code where we could verify the APIs produce better behaviour than right now? As a way to validate? 

We could even apply them here, just to be sure...

Reply:
Sure thing! [Here](https://github.com/liferay/liferay-portal/blob/ef37548663a6635f3be7e8596554c8b1e32c869e/modules/apps/layout/layout-taglib/src/main/java/com/liferay/layout/taglib/internal/display/context/RenderLayoutStructureDisplayContext.java#L863-L881) and [here](https://github.com/liferay/liferay-portal/blob/81fa71c00ea65687f78daffedb903db57f5c47c5/modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/util/StyleBookEntryUtil.java#L40-L62)